task_runner.py

```diff
@@ -11,19 +11,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# pylint: disable=unused-import
+# pylint: disable=useless-return
+# pylint: disable=protected-access
+# pylint: disable=bare-except
+# pylint: disable=global-statement
+
+
 import os
-import sys
+import signal
 import subprocess
+import sys
+import time
 
-import adcm.init_django		# pylint: disable=unused-import
-
-from django.utils import timezone
+import adcm.init_django
 from django.core.exceptions import ObjectDoesNotExist
-from cm.models import TaskLog, JobLog
+from django.utils import timezone
 
-from cm.logger import log
 import cm.config as config
 import cm.job
+from cm.logger import log
+from cm.models import TaskLog, JobLog
+
+
+TASK_ID = 0
+
+
+def terminate_job(task, jobs):
+    running_job = jobs.get(status=config.Job.RUNNING)
+
+    if running_job.pid:
+        os.kill(running_job.pid, signal.SIGTERM)
+        cm.job.finish_task(task, running_job, config.Job.ABORTED)
+    else:
+        cm.job.finish_task(task, None, config.Job.ABORTED)
+
+
+def terminate_task(signum, frame):
+    log.info("cancel task #%s, signal: #%s", TASK_ID, signum)
+    task = TaskLog.objects.get(id=TASK_ID)
+    jobs = JobLog.objects.filter(task_id=TASK_ID)
+
+    i = 0
+    while i < 10:
+        if jobs.filter(status=config.Job.RUNNING):
+            terminate_job(task, jobs)
+            break
+        i += 1
+        time.sleep(0.5)
+
```

Comment:
we need to call cm.job.finish_task() here in case we didn't find any running job

Reply:
Ok, thanks.